### ISSUE TITLE
PATCH: inconsistent Add operation behavior.

### ISSUE THREAD

**Issue by IlyaAbramau** [simpleidserver/SimpleIdServer#359]
According to the RFC documentation (Add operation) https://www.rfc-editor.org/rfc/rfc7644#section-3.5.2.1:
"If the target location already contains the value specified, **no changes SHOULD be made to the resource, and a success response SHOULD be returned**.  Unless other operations change the resource, this operation SHALL NOT change the modify timestamp of the resource."

At the current moment if we make this operation: the library **do changes**.
Please, look at PatchRepresentationCommandHandler.cs:
![image](https://user-images.githubusercontent.com/118731468/206196343-7b1b468a-1fc9-40a3-b8be-d90338e57e18.png)
If we trying to update the resource property with the same value _patchResult_ contains "Remove" command instead empty list of commands.

The expected behavior is _patchResult_ contains empty list.

The possible solution can be looked like:
SCIMRepresentationExtensions.cs  -> ApplyPatches method

<img width="911" alt="image" src="https://user-images.githubusercontent.com/118731468/206198216-e7cd295c-1672-4838-a839-07ff9e3998de.png">

the 207 row  removeCallback is always called, even when newAttributes has empty list <- here we are getting only Remove command as a result. We can add an additional filtering by newAttributes.FullPath when removeCallback  is called:

```
     var newAttributes = ExtractRepresentationAttributesFromJSON(representation.Schemas, schemaAttributes.ToList(), patch.Value, ignoreUnsupportedCanonicalValues);
     newAttributes = RemoveStandardReferenceProperties(newAttributes, attributeMappings);
     newAttributes = FilterDuplicate(attributes, newAttributes);
      // here changes:
     var newAttributesPathsDictionary = newAttributes.Select(s => s.FullPath).ToDictionary(x => x, x => x);
     removeCallback(attributes.Where(
                                a => !a.SchemaAttribute.MultiValued &&
                                a.FullPath == fullPath &&
                                newAttributesPathsDictionary.ContainsKey(a.FullPath)).ToList());
     ....
```
The idea is - if here is nothing to update (newAttributes list is empty) we are not applied any commands (remove).
 So, the whole method will return empty list and PatchRepresentationCommandHandler.UpdateRepresentation will return control without modifying the resource.             

_**I tested it with newAttributes count = 1, please make sure that it works properly for several attributes for the newAttributes variable (if it available).**_

**Comment by simpleidserver** [comment]
Hello,

First thank you for you feedback :)
The ticket is a duplicate of this one #358 and is already fixed in the branch `release/v2.0.22`.

KR,

SID

**Comment by IlyaAbramau** [comment]
Issue is fixed, thank you.
